Public/Src/Tools/Execution.Analyzer/Analyzers.Partner/ContentPlacementAnalyzer.cs

```diff
@@ -0,0 +1,532 @@
+                            AddToDictionaryWithListAsValue(m_artifactOutputPips, output.FileArtifact, pip);
+                            CountInDictionary(m_scheduledPipOutputCount, pip.PipId.Value);
+                        }
+
+                        return true;
+                    },
+                    includeUncacheable: true);
+                    // the dependencies, cause we want to count them...
+                    foreach (var dependency in CachedGraph.PipGraph.RetrievePipImmediateDependencies(pip))
+                    {
+                        CountInDictionary(m_scheduledPipDependencyCount, pip.PipId.Value);
+                    }
+                    ++totalPips;
+                }
+
+            }
+            Console.WriteLine($"Total artifacts is {m_fileContentMap.Count}, sampling (max possible samples = {m_sampleSizeHardLimit})");
+            // now, for each file artifact...
+            int processed = 0;
+            int errors = 0;
+            // sample
+            var samples = SampleAccordingToOverallDistribution(m_sampleSize, new Random(Environment.TickCount));
+            int printStep = samples.Count > 100000 ? 100000 : 1000;
+            // go ahead...
+            Console.WriteLine($"{samples.Count} after sampling ({m_fileContentMap.Count} total), printing...");
+            try
+            {
+                m_outputWriter.WriteStartArray();
+                foreach (var artifactEntry in samples)
+                {
+                    try
+                    {
+                        var artifact = artifactEntry.Key;
+                        var artifactInfo = artifactEntry.Value;
+                        var bxlArtifact = new BXLArtifact
+                        {
+                            Hash = artifactInfo.Hash.ToHex(),
+                            ReportedFile = Path.GetFileName(artifact.Path.ToString(CachedGraph.Context.PathTable)),
+                            ReportedSize = artifactInfo.Length,
+                            BuidId = m_buildId,
+                            BuildQueue = m_buildQueue,
+                            BuildDurationMs = m_buildDurationMs,
+                            BuildStartTimeTicks = m_buildStartTicks,
+                            TotalPips = totalPips
+                        };
+
+                        var inputPipsForThisArtifact = GetOrReturnNull(m_artifactInputPips, artifact);
+                        var outputPipsForThisArtifact = GetOrReturnNull(m_artifactOutputPips, artifact);
+                        // and process this entry
+                        if (inputPipsForThisArtifact != null)
+                        {
+                            foreach (var artifactPip in inputPipsForThisArtifact)
+                            {
+                                IList<PipExecutionPerformanceEventData> pipExecutionData;
+                                if (m_scheduledPipExecutionData.TryGetValue(artifactPip.PipId.Value, out pipExecutionData))

```

Comment:
>out pipExecutionData [](start = 101, length = 20)

similar to a comment earlier. You can use
`out var pipExecutionData` to reduce the verbosity of the code slightly.